src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -78,6 +79,44 @@ public void createPrimaryUserAffiliation(String eventPayload) {
     }
   }
 
+  @Override
+  @SneakyThrows
+  @Transactional
+  public void updatePrimaryUserAffiliation(String eventPayload) {
+    FolioExecutionContext currentContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    String centralTenantId = folioExecutionContext.getTenantId();
+    try {
+      var userEvent = objectMapper.readValue(eventPayload, UserEvent.class);
+      log.info("Received event for update primary affiliation for user: {} and tenant: {}", userEvent.getUserDto().getId(), userEvent.getTenantId());
+
+      var consortiaTenant = tenantService.getByTenantId(userEvent.getTenantId());
+      if (consortiaTenant == null) {
+        log.warn("Tenant {} not exists in consortia", userEvent.getTenantId());
+        return;
+      }
+      UUID userId = getUserId(userEvent);
+      String newUsername = userEvent.getUserDto().getUsername();
+
+      UserTenantCollection userTenantCollection = userTenantService.getByUserId(consortiaTenant.getConsortiumId(), userId, 0, Integer.MAX_VALUE);
```

Comment:
We should use method: findByUserIdAndTenantId, it already exists in DAO, need to add it it UseTenantService and use ,
tenantId we should use from eventPayload

Reply:
Refactored